features/285-design-webhooks/events.md

```diff
@@ -0,0 +1,260 @@
+# Web-hook Events
```

Comment:
@kiootic @louischan-oursky Have we looked at our list of all current auth API, and make sure use cases are covered?

(Actually, where is the current latest list of all auth APIs? I want to look at it and make some comparisons myself...)

I notice at least one hook is not covered -- after password change. We had a real use case before: https://github.com/SkygearIO/features/issues/206

Can we add the above back? (and also make sure if the whole design still make sense) And also review and see if the web hooks are probably designed with common use cases.

Reply:
Yes I've considered password changed event before. However, since we'd like to implement the minimum required amount of events at the moment due to schedule, so I left it out.

Reply:
> (Actually, where is the current latest list of all auth APIs? I want to look at it and make some comparisons myself...)

I always want to consolidate all auth gear features into a single document so that it is easier for us to see what auth gear currently offers. I also want to archive obsolete or outdated documents so the repo reflects the implemented feature and behavior.

---

https://github.com/SkygearIO/features/tree/master/features/323-login-id

This document covers signup and login.

---

https://github.com/SkygearIO/features/pull/334

This document covers add/remove/replace login id, change password, list identities and the auth response shape.

---

https://github.com/SkygearIO/features/tree/master/features/332-sso

This documents covers SSO, including OAuth 2.0, Custom Token and external access token.
This documents obsoletes

- https://github.com/SkygearIO/features/tree/master/features/5-basic-sso
- https://github.com/SkygearIO/features/tree/master/features/56-login-with-custom-token

---

User verification is not specified in next. Though we have the following documents from v1

- https://github.com/SkygearIO/features/tree/master/features/20-sms-verification
- https://github.com/SkygearIO/features/tree/master/features/22-email-verification

---

Disabling user is not specified in next. Though we have the following document from v1

- https://github.com/SkygearIO/features/tree/master/features/50-disable-user

---

There are some next documents that may be outdated

- https://github.com/SkygearIO/features/tree/master/features/253-auth-sdk-callback-in-next
- https://github.com/SkygearIO/features/tree/master/features/256-auth-gear-interaction-overview-in-next
- https://github.com/SkygearIO/features/tree/master/features/285-design-webhooks
- https://github.com/SkygearIO/features/tree/master/features/294-login-without-specifying-login-id-key

--- 

The following documents are record related so they should be irrelevant now.

- https://github.com/SkygearIO/features/tree/master/features/48-user-profile-discovery

Reply:
> Yes I've considered password changed event before. However, since we'd like to implement the minimum required amount of events at the moment due to schedule, so I left it out.

Ic. It is okay to create spec which doesn't implement yet. Usually, our flows are:

1. Create feature issues
2. Write the high level requirements / APIs
3. Create spec for the detailed user interface
4. Create repo's issues, put those repo issues number as link in feature issue
5. Implement
6. Complete API docs
7. Release
8. Close feature issues

So don't worry about writing a complete spec but implement only a portion of it.

On the other hand, we can also open feature issues for bigger features that seems too large to mix together.

Reply:
> I always want to consolidate all auth gear features into a single document so that it is easier for us to see what auth gear currently offers. I also want to archive obsolete or outdated documents so the repo reflects the implemented feature and behavior.
> 
> [](https://github.com/SkygearIO/features/tree/master/features/323-login-id)

@louischan-oursky sure, go ahead... Actually it is good if we can consolidate the "releasable version" of Skygear Auth's remaining features / spec in the original tracking feature issue: https://github.com/SkygearIO/features/issues/256